app/channel_test.go

```diff
@@ -2418,13 +2390,13 @@ func TestMarkUnreadWithThreads(t *testing.T) {
 			appErr = th.App.UpdateThreadFollowForUser(th.BasicUser.Id, th.BasicTeam.Id, rootPost.Id, false)
 			require.Nil(t, appErr)
 
-			_, appErr = th.App.MarkChannelAsUnreadFromPost(rootPost.Id, th.BasicUser.Id, true, true)
+			_, appErr = th.App.MarkChannelAsUnreadFromPost(rootPost.Id, th.BasicUser.Id, true)
 			require.Nil(t, appErr)
 
 			threadMembership, appErr := th.App.GetThreadMembershipForUser(th.BasicUser.Id, rootPost.Id)
 			require.Nil(t, appErr)
 			require.NotNil(t, threadMembership)
-			assert.Equal(t, int64(1), threadMembership.UnreadMentions)
+			assert.Zero(t, threadMembership.UnreadMentions)
```

Comment:
@ashishbhate, just want to get your eyes on this -- and the comment newly introduced at the top of this method.

In short, the coverage here was excellent -- but of an unused code path. I didn't want to refactor `MarkChannelAsUnreadFromPost` further at this time -- in which I /think/ we can rip out all the `ThreadStore.MaintainMembership` calls -- so opted to "correct" this test accordingly.

Reply:
@lieut-data Sorry about the delay in getting around to this. 

I think the change here breaks the expected behaviour. `App.MarkChannelAsUnreadFromPost`  is called from the `rhs_root_post` and `rhs_comment` components in the webapp (amongst other places). 

Looking at our spreadsheet of expected behaviour - https://docs.google.com/spreadsheets/d/1Rp2UN-TDW75zccfFWrco2Pqog2ktoa784GSvAVGTfgA/edit?pli=1#gid=1722303699

The test marks the root post as unread - this is covered by lines 17 and 18, which states that a new messages line should be inserted before the oldest post.

The reply in the test has a mention, so inserting a new messages line should also increment the mention count for the user. 

@koox00 could you please confirm if this is correct? Two eyes are better than one :)

Reply:
@lieut-data I think this comment was missed in the review. This PR appears to break existing  expected behaviour. Jelena opened a ticket for issue, but if this PR is the cause of the regression then we ought to prioritize the fix. 

// @Willyfrog